src/openforms/submissions/tests/test_variables/test_variable_injection.py

```diff
@@ -221,3 +221,122 @@ def test_logic_evalution_endpoint(self):
                 "content1",
                 {"html": "<p>some input enabled: no</p>"},
             )
+
+
+class VariableInjectionI18NTests(SubmissionsMixin, FormioMixin, APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.submission = SubmissionFactory.from_components(
+            language_code="en",
+            components_list=[
+                {
+                    "key": "naam",
+                    "type": "textfield",
+                    "label": "Naam",
+                },
+                {
+                    "key": "geboortedatum",
+                    "type": "date",
+                    "label": "Geboortedatum",
+                    "format": "dd-MM-yyyy",
+                },
+                {
+                    "key": "ww",
+                    "type": "password",
+                    "label": "Wachtwoord",
+                    "description": 'Suggestie: #{{naam|title}}{{geboortedatum|date:"Y"}}',
+                },
+                {
+                    "key": "ww_check",
+                    "type": "checkbox",
+                    "label": "Bevestig uw wachtwoord: {{ww}}",
+                    "description": "Mooi hè? We laten u niet 2 maal hetzelfde "
+                    "typen. Gemak > veiligheid.",
+                },
+            ],
+            submitted_data={
+                "naam": "Alice",
+                "geboortedatum": "1865-01-01",
+                "ww": "L3GyL4V5Maus2Jg",
+            },
+        )
+        cls.form_step = cls.submission.steps[0].form_step
+        cls.form_step.form_definition.component_translations = {
+            "en": {
+                'Suggestie: #{{naam|title}}{{geboortedatum|date:"Y"}}': 'Suggestion: #{{naam|title}}{{geboortedatum|date:"Y"}}',
+                "Naam": "Name",
+                "Geboortedatum": "Birthdate",
+                "Bevestig uw wachtwoord: {{ww}}": "Confirm your password: {{ww}}",
+                # Actually en-ca, eh?
+                "Mooi hè? We laten u niet 2 maal hetzelfde typen. Gemak > veiligheid.": "Nice, eh? We won't make you type the same thing twice. Convenience over safety.",
+            }
+        }
+        cls.form_step.form_definition.save()
+
+    def test_variable_interpolates_on_translated_definitions(self):
+        self._add_submission_to_session(self.submission)
+        endpoint = reverse(
+            "api:submission-steps-detail",
+            kwargs={
+                "submission_uuid": self.submission.uuid,
+                "step_uuid": self.form_step.uuid,
+            },
+        )
+
+        step_json = self.client.get(endpoint).json()
+        components_by_key = {
+            c["key"]: c for c in step_json["formStep"]["configuration"]["components"]
+        }
+
+        c = components_by_key.get
```

Comment:
a horrible variable/function name when in `pdb` :sweat_smile:

Reply:
pip install pdbpp and never look back.